server/plugin/taskLinkPreview.go

```diff
@@ -25,14 +26,81 @@ func (p *Plugin) PostTaskPreview(linkData []string, userID, channelID string) (*
 		description = "No description"
 	}
 
-	taskTitle := fmt.Sprintf(constants.TaskTitle, task.Fields.Type, task.ID, task.Fields.Title, task.Link.HTML.Href)
-	TaskPreviewMessage := fmt.Sprintf(constants.TaskPreviewMessage, task.Fields.State, assignedTo, description)
 	post := &model.Post{
 		UserId:    userID,
 		ChannelId: channelID,
 	}
 	attachment := &model.SlackAttachment{
-		Text: fmt.Sprintf("%s\n%s\n", taskTitle, TaskPreviewMessage),
+		AuthorName: "Azure Boards",
+		AuthorIcon: fmt.Sprintf("%s/plugins/%s/static/%s", p.GetSiteURL(), constants.PluginID, constants.BoardsIcon),
+		Title:      fmt.Sprintf(constants.TaskTitle, task.Fields.Type, task.ID, task.Fields.Title, task.Link.HTML.Href),
+		Color:      constants.BoardsIconColor,
+		Fields: []*model.SlackAttachmentField{
+			{
+				Title: "State",
+				Value: task.Fields.State,
+				Short: true,
+			},
+			{
+				Title: "Assigned To",
+				Value: assignedTo,
+				Short: true,
+			},
+			{
+				Title: "Description",
+				Value: description,
+			},
+		},
+		Footer:     linkData[4],
+		FooterIcon: fmt.Sprintf("%s/plugins/%s/static/%s", p.GetSiteURL(), constants.PluginID, constants.ProjectIcon),
+	}
+	model.ParseSlackAttachment(post, []*model.SlackAttachment{attachment})
+	return post, ""
+}
+
+func (p *Plugin) PostPullRequestPreview(linkData []string, link, userID, channelID string) (*model.Post, string) {
+	pullRequest, _, err := p.Client.GetPullRequest(linkData[3], linkData[8], linkData[6], userID)
```

Comment:
is there a check for the length of linkData if there is nice if not please include this

Reply:
We don't have any direct check but we are using regex exp to check for the link and if the regex is passed then it definitely has this length